uk.co.compucorp.civicrm.hrcore/backstop_data/page-objects/ssp-leave-absences-manager-leave-requests.js

```diff
@@ -0,0 +1,84 @@
+var Promise = require('es6-promise').Promise;
+var page = require('./page');
+
+module.exports = (function () {
+
+  return page.extend({
+    /**
+     * Wait for the page to be ready
+     * @return {Object} this object
+     */
+    waitForReady: function () {
+      var casper = this.casper;
+
+      casper.then(function () {
+        casper.waitUntilVisible('tbody  tr:nth-child(1) a');
+      });
+
+      return this;
+    },
+    /**
+     * Opens the given menu item from side menu
+     * @param {String} menuIndex
+     * @return {Object} this object
+     */
+    openSideMenu: function (menuIndex) {
+      var casper = this.casper;
+
+      casper.then(function () {
+        casper.waitUntilVisible('ul.chr_vertical_tabs li:nth-child(1) a', function () {
```

Comment:
why are you always waiting for something before you even begin the action? That initial waiting period should be taken care of by `waitForReady`